src/core/pratt.ml

```diff
@@ -0,0 +1,133 @@
+(** A simple parser for operators. See Pratt parsing or the Shunting Yard
+    algorithm.
+
+    <https://dev.to/jrop/pratt-parsing>
+    <https://effbot.org/zone/simple-top-down-parsing.htm> *)
+
+open Lplib
+open Lplib.Extra
+
+open Syntax
+
+(** Type of operators with additional data. *)
+(* REVIEW: simplify the type into [Unary of float * assoc] and remove [unop]
+   and [binop] *)
+type operator =
+  | Unary of unop
+  | Binary of binop
+
+(** Table containing all registered binary and unary operators that may appear
+    in terms parsed by {!val:Pratt.expression}. *)
+let operators : operator StrHtbl.t = StrHtbl.create 17
```

Comment:
This map should evolve with the state (Timed). In fact, pratt should depend on the sig_state which already contains the operator map in the field pp_hints.

Reply:
I do completely agree with that. In fact, I'm trying to externalise completely the Pratt part, so that we just call a library. And the operator are in the `signature` of the `sig_state`, right?

Reply:
Ok but the point is about the correct update of the operators table. Why not removing operators and passing pp_hints as argument to the pratt functions?

Reply:
The operators are in the pp_hints filed of the sig_state, not in the signature.